### PR TITLE
Feature/3095  content type in event type

### PR DESCRIPTION
**Pull request by grignards**
Adds an optional sub-type for events. This sub-type is used in content events to indicate the content's type.

Refs #3095.

## Checkpoints

The goal of this section is to help code reviewers and the merge request author to do the required checks on this pull request. Please don't edit this list.
If one or more of these checkpoints are out of scope, please write below why they are.

**For code reviewers**

- [x] The code is clear enough
- [x] If there are FIXMEs in the code, then there are associated issues and issue is referenced in the FIXME
- [x] If there are TODOs, NOTEs or HACKs in code, the date and the developer initials are present
- [x] Automated tests have been written (covering the feature or non regression), or *at least* an issue has been created to implement the test

**For developers**

- [x] Manual tests done to ensure stability on whole application layers, issue expectation follow
- [x] The original issue has been updated with a short summary of the implemented solution

**For the quality team**

- [x] Tested by the quality team

### CODE REVIEW

File: backend/tests_configs.ini [hunk 1]
```diff
@@ -202,6 +202,7 @@ email.notification.enabled_on_invitation = False
 webdav.ui.enabled = False
 webdav.base_url = https://localhost:3030
 webdav.root_path = /webdav
+debug = True
```
Comment (inkhey) [inline backend/tests_configs.ini:205]: Should be removed here no ?

File: backend/tracim_backend/migration/versions/a074395d3caf_add_subtype_in_events_table.py [hunk 2]
```diff
@@ -0,0 +1,26 @@
...
+# revision identifiers, used by Alembic.
+revision = "a074395d3caf"
+down_revision = "3c92f57c52b2"
+
+
+def upgrade():
+    with op.batch_alter_table("events") as batch_op:
+        batch_op.add_column(sa.Column("entity_subtype", sa.String(), nullable=True))
+    # NOTE SG 2020-05-26: subtype values are left to NULL as we do not have any production
+    # database running with the events table.
```
Comment (inkhey) [inline backend/tracim_backend/migration/versions/a074395d3caf_add_subtype_in_events_table.py:21]: What does this imply for developer database that used the previous implementation of TLM, and do try to use the same database updated ?

Comment (grignards) [reply backend/tracim_backend/migration/versions/a074395d3caf_add_subtype_in_events_table.py:21]: This imply that content events/messages previously saved in the database won't have their content-type added when queried through the HTTP API.
As we only have some developer databases, the easier way is for them to delete/recreate their database.

Production database migration will be OK as the events table didn't exist in 2.7.x.

File: backend/tests_configs.ini (hunk 1 above)
Comment (grignards) [reply backend/tests_configs.ini:205]: Again I forgot it… Is there a way to set this config parameter through an environment variable ? I find it quite useful to get the backtrace when an http api responds a 500 error.

Comment (inkhey) [reply backend/tests_configs.ini:205]: Yes, like most parameters of Tracim.
Set TRACIM_DEBUG.

Comment (grignards) [reply backend/tests_configs.ini]: OK, thank your for the tip.

File: backend/tracim_backend/models/event.py [hunk 3]
```diff
@@ -70,7 +70,7 @@ class Event(DeclarativeBase):
     event_id = Column(Integer, autoincrement=True, primary_key=True)
     operation = Column(Enum(OperationType), nullable=False)
     entity_type = Column(Enum(EntityType), nullable=False)
-    entity_subtype = Column(String, nullable=True, default=None)
+    entity_subtype = Column(String(length=100), nullable=True, default=None)
```
Comment (inkhey) [inline backend/tracim_backend/models/event.py:73]:
```suggestion
    entity_subtype = Column(String(length=ENTITY_SUBTYPE_LENGTH), nullable=True, default=None)
```
may be better.

**Comment by raphj** [comment]
Can you add a small sentence that is fast to read at the beginning of the issue? (Imagine writing the changelog for the release and browsing the tickets to decide what to put in)

**Comment by grignards** [comment]
> lgtm.
> 
> We are sure the Content type will never be mistaken for the MIME type (the Content-Type HTTP header), right?

We have a `file_mimetype` field in contents for this, `content_type` is already existing in the HTTP API so the confusion is not too likely.

**Comment by grignards** [comment]
> Can you add a small sentence that is fast to read at the beginning of the issue? (Imagine writing the changelog for the release and browsing the tickets to decide what to put in)

Done.

**Comment by grignards** [comment]
No need for quality testing as this is only a backend development for which the frontend development is still ongoing.

**Comment by inkhey** [comment]
Cypress test broken due to missing fix in develop (available in another branch), merge this to deblock frontend dev.

**Pull request merged**